src/content/docs/dev/general-message-passing/soroban-gmp/soroban-contracts.mdx

```diff
@@ -0,0 +1,128 @@
+# Soroban Contracts
+
+The core contracts for Axelar's integration with Stellar Soroban can be found at the [axelar-cgp-soroban](https://github.com/axelarnetwork/axelar-cgp-soroban) repository. 
+
+The Rust based GMP contracts can be thought of similarly to their [EVM](/dev/general-message-passing/gmp-messages/) counterparts. 
+
+At the core of there are two main contracts involved in sending a GMP message, these are the [Gateway Contract](https://github.com/axelarnetwork/axelar-cgp-soroban/tree/main/contracts/axelar-gateway) and the [Gas Service](https://github.com/axelarnetwork/axelar-cgp-soroban/tree/main/contracts/axelar-gas-service).
+
+## Soroban Gateway
+The Gateway facilitates sending and receiving of cross-chain messages to other chains via the Axelar Network.
+
+For sending a GMP message, the `callContract()` function needs to be triggered.
+
+### CallContract
+The `callContract` function triggers your cross-chain message from Stellar to the Axelar Network. When sending a cross-chain message you must specify the destination of your cross-chain message and a given payload.
+    
+The `callContract` function takes five parameters. 
+1. `env`: Standard soroban [env](https://developers.stellar.org/docs/learn/encyclopedia/contract-development/environment-concepts) that provides access to the environment the contract is executing within. 
```

Comment:
Not sure it's a particularly relevant detail, but perhaps worth noting that the Env is not explicitly passed to the client.

Reply:
Good call will do